libr/core/cmd_print.c

```diff
@@ -1330,9 +1345,8 @@ static int pdi(RCore *core, int nb_opcodes, int nb_bytes, int fmt) {
 	int i=0, j, ret, err = 0;
 	ut64 old_offset = core->offset;
 	RAsmOp asmop;
-	#define PAL(x) (core->cons && core->cons->pal.x)? core->cons->pal.x
-	const char *color_reg = PAL(reg): Color_YELLOW;
-	const char *color_num = PAL(num): Color_CYAN;
+	const char *color_reg = pdi_get_color(core, offsetof(RConsPalette, reg), Color_YELLOW);
```

Comment:
Ideally colors should be stored in an array rather than a struct to allow selecting color by index. As for offsetoff, I'll change it, but it's a part of C standard.